File: Data Structures & Algorithms/length-of-last-word/submission-3.kt
class Solution {
    fun lengthOfLastWord(s: String): Int {
        var r = s.length - 1
        while(r >= 0 && s[r] == ' ') {
            r--
        }
        
        var c = 0
        while(r >= 0 && s[r] != ' ') {
            c++
            r--
        }

        return c
    }
}

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-3.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        prefix = [1] * len(nums)
        postfix = [1] * len(nums)

        for i in range(len(nums)):
            if i == 0:
                prefix[i] = nums[i]
            else:
                prefix[i] = nums[i] * prefix[i - 1]
        
        for i in range(len(nums) - 1, -1, -1):
            if i == len(nums) - 1:
                postfix[i] = nums[i]
            else:
                postfix[i] = nums[i] * postfix[i + 1]
        
        result = [1] * len(nums)
        for i in range(len(nums)):
            pre = prefix[i - 1] if i > 0 else 1
            post = postfix[i + 1] if i < len(nums) - 1 else 1
            result[i] = pre * post

        return result